contrib/confluentinc/confluent-kafka-go/kafka.v2/option.go

```diff
@@ -0,0 +1,106 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package kafka
+
+import (
+	"context"
+	"math"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/internal"
+	"gopkg.in/DataDog/dd-trace-go.v1/internal/namingschema"
+
+	"github.com/confluentinc/confluent-kafka-go/v2/kafka"
+)
+
+type config struct {
+	ctx                   context.Context
+	consumerServiceName   string
+	producerServiceName   string
+	consumerOperationName string
+	producerOperationName string
+	analyticsRate         float64
+	tagFns                map[string]func(msg *kafka.Message) interface{}
+}
+
+// An Option customizes the config.
+type Option func(cfg *config)
+
+func newConfig(opts ...Option) *config {
+	cfg := &config{
+		ctx: context.Background(),
+		// analyticsRate: globalconfig.AnalyticsRate(),
+		analyticsRate: math.NaN(),
+	}
+	if internal.BoolEnv("DD_TRACE_KAFKA_ANALYTICS_ENABLED", false) {
+		cfg.analyticsRate = 1.0
+	}
+
+	cfg.consumerServiceName = namingschema.NewServiceNameSchema("", "kafka").GetName()
+	cfg.producerServiceName = namingschema.NewServiceNameSchema(
+		"",
+		"kafka",
+		namingschema.WithVersionOverride(namingschema.SchemaV0, "kafka"),
+	).GetName()
+
+	cfg.consumerOperationName = namingschema.NewKafkaInboundOp().GetName()
+	cfg.producerOperationName = namingschema.NewKafkaOutboundOp().GetName()
+
+	for _, opt := range opts {
+		opt(cfg)
+	}
+	return cfg
+}
+
+// WithContext sets the config context to ctx.
+// Deprecated: This is deprecated in favor of passing the context
+// via the message headers
+func WithContext(ctx context.Context) Option {
```

Comment:
I don't think we need to add this deprecated option to the new integration, since people will have to update their code anyway. Any thoughts? @ajgajg1134 @ahmed-mez

Reply:
@knusbaum